grazer/src/main/java/edu/asu/diging/grazer/core/graphs/impl/GraphManager.java

```diff
@@ -53,29 +54,48 @@
     private List<String> transformationNames;
     private Cache cache;
     
+    private File[] fileNames;
+    
     @PostConstruct
     public void init() {
-        transformationNames = new ArrayList<>();
-        transformationNames.add(PERSON_SIMPLE_TRIPLE);
-        transformationNames.add(PERSON_OBJECT_SIMPLE_TRIPLE);
-        transformationNames.add(PERSON_HAS_SOMEONE);
-        transformationNames.add(SOMEONE_HAS_PERSON);
-        transformationNames.add("someone_has_sth_start_date");
-        transformationNames.add("sth_has_so_start_date");
-        transformationNames.add("someone_has_sth_end_date");
-        transformationNames.add("sth_has_so_end_date");
-        transformationNames.add("someone_has_sth_occur_date");
-        transformationNames.add("sth_has_so_occur_date");
         
+        File folder = new File(folderName);
+    		transformationNames = new ArrayList<>();        
+        
+    		FileFilter filter = new FileFilter() {
+    			public boolean accept(File file) {
+    				if (file.getName().endsWith(fileExtension) && file.getName().startsWith(prefix)) {
+    					return true;
+    				}
+    				return false;
+    			}
+    		};
+    		
+    		if(folder.exists()) {
+    			if(folder.isDirectory()) {
+	        		fileNames = folder.listFiles(filter);
+	        		if(fileNames != null && fileNames.length > 0) {
+	        			for(int i = 0; i < fileNames.length; i++) {
+	        				// Removing prefixes PAT_ and TRA_ and .graphml at the end 
+		        			String file = fileNames[i].getName().substring(4).replaceFirst("[.][^.]+$", "");
+		        			if(!transformationNames.contains(file)) {
```

Comment:
you probably don't need this check (there should not be duplicate filenames). if you want to make sure this can absolutely not happen (duplicates in the list), just use a Set instead of a list.